clients/src/main/java/org/apache/kafka/common/utils/Base64.java

```diff
@@ -209,8 +244,12 @@ public String encodeToString(byte[] bytes) {
                 //Convert to URL safe variant by replacing + and / with - and _ respectively.
                 String urlSafeBase64EncodedUUID = base64EncodedUUID.replace("+", "-")
                         .replace("/", "_");
-                // Remove the "==" padding at the end.
-                return urlSafeBase64EncodedUUID.substring(0, urlSafeBase64EncodedUUID.length() - 2);
+                // Remove any "=" or "==" padding at the end.
+                return urlSafeBase64EncodedUUID.endsWith("==")
+                        ? urlSafeBase64EncodedUUID.substring(0, urlSafeBase64EncodedUUID.length() - 2)
+                        : urlSafeBase64EncodedUUID.endsWith("=")
+                                ? urlSafeBase64EncodedUUID.substring(0, urlSafeBase64EncodedUUID.length() - 1)
+                                : urlSafeBase64EncodedUUID;
```

Comment:
@rondagostino Thanks for the PR. Couldn't we just use substring until the first "=" ? Something like this would be easier to read?
```
int index = urlSafeBase64EncodedUUID.indexOf('=');
return index > 0 ? urlSafeBase64EncodedUUID.substring(0, index) : urlSafeBase64EncodedUUID;
```

Reply:
I used `endsWith()` as a knee-jerk reaction to avoid scanning the entire string from the beginning, but now that you mention it I see that we can use still use the more clear `indexOf()` while avoiding a full scan by starting the scan near the end.  I'll push something like the following.  I'll also special-case (and test) the empty byte array. 
 Thanks.
```
int index = urlSafeBase64EncodedUUID.indexOf('=', urlSafeBase64EncodedUUID.length() - 2);
return index > 0 ? urlSafeBase64EncodedUUID.substring(0, index) : urlSafeBase64EncodedUUID;
```